File: Python/khudi/light/Light.java
package khudi.light;

import khudi.color.RGBColor;
import khudi.math.Vector;

/**
 * @section DESCRIPTION
 *
 * The Light Class.
 * Provides the Light class ....
 *
 */
public class Light
{
	private Vector point;
	private RGBColor color;

	public Light()
	{
	}

	public Light(Vector pointp, RGBColor colorp)
	{
		point = pointp;
		color = colorp;
	}

	public Light(Light light)
	{
		point = light.point;
		color = light.color;
	}

	public void SetPosition(Vector v)
	{
		point = v;
	}

	public Vector GetPosition()
	{
		return point;
	}

	public RGBColor GetColor()
	{
		return color;
	}

	public void SetColor(RGBColor colorp)
	{
		color = colorp;
	}

}

File: Python/khudi/light/Light.py
#!/usr/bin/env python
""" generated source for module Light """
from __future__ import print_function
# package: khudi.light
# 
#  * @section DESCRIPTION
#  *
#  * The Light Class.
#  * Provides the Light class ....
#  *
#  
class Light(object):
    """ generated source for class Light """
    point = None
    color = None

    def __init__(self, x=None, y=None):
        """ generated source for method __init__ """
        if isinstance(x, Light):
            self.point = x.point
            self.color = x.color
        else:
            self.point = x
            self.color = y


    def SetPosition(self, v):
        """ generated source for method SetPosition """
        self.point = v

    def GetPosition(self):
        """ generated source for method GetPosition """
        return self.point

    def GetColor(self):
        """ generated source for method GetColor """
        return self.color

    def SetColor(self, colorp):
        """ generated source for method SetColor """
        self.color = colorp

